### DISCUSSION TITLE
Indexed Color Drawing?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [alternetsoft/AlternetUI#135] [Ideas]
I don't know if true indexed color drawing is possible but can we have the ability to map colors to other colors?

**Comment by generalloki** [comment]
Where did you see this? Any code examples? Any suggestions on how this would look?

**Reply by Shadowblitz16** [reply]
public record ColorMap(Color Source, Color Target);
public record Palette(ColorMap[] colors);

**Comment by Shadowblitz16** [comment]
@generalloki winforms has support for color mapping with Palette and ColorMap
I was going to use it for yy-chr style graphics but supporting every old school game console graphics format would take your time out of improving alternate ui.


if your interested in supporting indexed graphics go look up yy-chr but even winforms doesn't support indexed graphics,
So I just suggested something like winforms colormap with drawing and images

**Comment by Shadowblitz16** [comment]
sorry accidentally closed it

**Comment by generalloki** [comment]
I am moving this to discussions.

**Comment by Shadowblitz16** [comment]
@generalloki maybe a MapColor event would be better? 
There needs to be some sort of support for effects.

**Comment by generalloki** [comment]
@Shadowblitz16 

We have added SKCanvas support (SkiaSharp). This canvas has many different features. Among them (I beleive) there is color mapping.

**Reply by Shadowblitz16** [reply]
@generalloki can we get compressed image drawing?
Meaning drawing image data less then 8 bits?

I would love to see support for drawing raw pixel data down to 1 bit per pixel without expanding it in memory. Palette data could be passed to the draw method instead of being part of the image data.

I ask this because color mapping only goes so far. 
I want to store super compressed image data.

**Reply by generalloki** [reply]
Our current strategy is the following: Provide some basic level painting via Graphics/Image. And suggest to use SkiaSharp for any advanced painting. I think SkiaSharp might support what you are seeking but I can not give any clues as never worked with 1 bit per pixel images.

**Reply by generalloki** [reply]
We do have Bitmap(int width, int height, int depth = 32) constructor. So you can probably pass depth 1 there and it will be bw image. But I am not aware how it will behave on different os and we did not perform any tests with such bitmaps. We just pass depth to wxWidgets and who knows how it works in case of bw bitmaps.
